### DISCUSSION TITLE
Vector search using partitions

### DISCUSSION THREAD

**Discussion by ArijitSinghEDA** [milvus-io/milvus#26491] [Q&A and General discussion] [answered]
I am conducting a vector search using something like this:
```
# Using Python 3.11

from pymilvus import Collection, Partition
import numpy as np

coll = Collection(name)  # Collection object having some partitions
partitions = None if len(coll.partitions) <= 1 else [d.name for d in coll.partitions]
for parts in coll.partitions:
    partobj = Partition(collection=coll, name=parts.name)
    partobj.load()
    embeddings = []
    ls = partobj.query(expr="id > -1", output_fields=["embeddings"])
    for dic in ls:
        embeddings.append(dic["embeddings"])
    smdata = np.array_split(embeddings, 10)
    for d in smdata:
        coll.search(
            data = d.tolist(),
            anns_field = "embeddings",
            param = {"metric_type": "IP"},
            limit = 2,
            expr = None,
            partition_names = partitions,
            output_fields = ["path"],
            round_decimal = 6,
            consistency_level = "Strong
        )
    partobj.release()
```

Now, I just have a simple doubt. Will this process search over the entire collection, or just the partition that is loaded?
My objective is to load one partition at a time and search its similar vectors over the entire collection. The reason is because I have a huge data (incremental in nature), and I cannot load it every time because of the loading limitation. If this method is wrong, can anyone suggest a workaround method?

**Comment by yhmo** [comment]
Your script wants to search the partitions one by one. No need to call `collection.search()`, you can search a partition by `partition.search()`.

```
from pymilvus import Collection, Partition
import numpy as np

coll = Collection(name)  # Collection object having some partitions
partitions = None if len(coll.partitions) <= 1 else [d.name for d in coll.partitions]
for parts in coll.partitions:
    partobj = Partition(collection=coll, name=parts.name)
    partobj.load()
    embeddings = []
    ls = partobj.query(expr="id > -1", output_fields=["embeddings"])
    for dic in ls:
        embeddings.append(dic["embeddings"])
    smdata = np.array_split(embeddings, 10)
    for d in smdata:
        partobj.search(
            data = d.tolist(),
            anns_field = "embeddings",
            param = {"metric_type": "IP"},
            limit = 2,
            expr = None,
            output_fields = ["path"],
            round_decimal = 6,
            consistency_level = "Strong",
        )
    partobj.release()
```

**Reply by ArijitSinghEDA** [reply]
Thanks for the reply! 
The thing is my collection is too large to get all the embeddings in one go. What I want to do is:

1. Get the embeddings from one partition
2. Search over all the collection
3. Iterate this process for every existing partition

Is there any way to do this without loading the entire collection? I am using Milvus Standalone v2.2.6

**Reply by yhmo** [reply, marked as answer]
Possible solution:
1. retrieve embeddings from partitions(load one by one), store the embeddings into a local file(numpy or json, etc)
load p1 ---> query p1 ----> store embeddings --> release p1 ---> load p2 ---> query p2 ---> store embeddings ---> release p2 ---> load p3 .....

2. search the embeddings batch by batch in a partition(load one by one)
load p1 ----> search(batch by batch) in p1---> release p1 ---> load p2 ---> search(batch by barch) in p2 ---> release p2 ---> load p3......

**Reply by ArijitSinghEDA** [reply]
Thank you! This should work!
